projects/batfish/src/main/java/org/batfish/bdp/AbstractRib.java

```diff
@@ -78,6 +85,68 @@ public void collectRoutes(Set<R> routes) {
       routes.addAll(_routes);
     }
 
+    public boolean containsRoute(
+        R route, BitSet bits, int prefixLength, int firstUnmatchedBitIndex) {
+      if (prefixLength == _prefix.getPrefixLength()) {
+        return _routes.contains(route);
+      } else {
+        boolean currentBit = bits.get(firstUnmatchedBitIndex);
+        if (currentBit) {
+          if (_right == null) {
+            return false;
+          } else {
+            Prefix rightPrefix = _right._prefix;
+            int rightPrefixLength = rightPrefix.getPrefixLength();
+            Ip rightAddress = rightPrefix.getAddress();
+            BitSet rightAddressBits = rightAddress.getAddressBits();
+            int nextUnmatchedBit;
+            boolean currentAddressBit = false;
+            boolean currentRightAddressBit;
+            for (nextUnmatchedBit = firstUnmatchedBitIndex + 1;
+                nextUnmatchedBit < rightPrefixLength && nextUnmatchedBit < prefixLength;
+                nextUnmatchedBit++) {
+              currentAddressBit = bits.get(nextUnmatchedBit);
+              currentRightAddressBit = rightAddressBits.get(nextUnmatchedBit);
+              if (currentRightAddressBit != currentAddressBit) {
+                break;
```

Comment:
Trying to see how to simplify this code a bit, if possible.

1. It looks to me like the `_right` and `_left` bits are identical except for whether you use `_right` or `_left`. Maybe extract a helper, and then this function could be simplified to (basically)

```java
if (currentBit) {
    doIt(_right);
} else {
    doIt(_left);
}
```

2. Can the `break` on this line just be `return false`, and then drop the `if` after the `for`?

Reply:
(p.s. I'm doubtful on 2. ; but see the comment above for why.)

Reply:
As mentioned above, I'd prefer to defer. Helpers were avoided due to criticality of this code. IMO any proposed changes that execute more code should be justified with profiling. However early exits are uncontroversial.